lib/ansible/plugins/inventory/now.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+DOCUMENTATION = r'''
+    name: now
+    plugin_type: inventory
+    author:
+      - Will Tome (@willtome)
+      - Alex Mittell (@alex_mittell)
+    short_description: ServiceNow Inventory Plugin
+    version_added: "2.10"
+    description:
+        - ServiceNow Inventory plugin
+    extends_documentation_fragment:
+        - constructed
+        - inventory_cache
+    options:
+        plugin:
+            description: The name of the ServiceNow Inventory Plugin, this should always be 'now'.
+            required: True
+            choices: ['now']
+        instance:
+            description: The ServiceNow instance URI. The URI should be the fully-qualified domain name, e.g. 'your-instance.servicenow.com'.
+            type: string
+            required: True
+            env:
+                - name: SN_INSTANCE
+        username:
+            description: The ServiceNow user acount, it should have rights to read cmdb_ci_server (default), or table specified by SN_TABLE
+            type: string
+            required: True
+            env:
+                - name: SN_USERNAME
+        password:
+            description: The ServiceNow instance user password.
+            type: string
+            secret: true
+            env:
+                - name: SN_PASSWORD
+        table:
+            description: The ServiceNow table to query
+            type: string
+            default: cmdb_ci_server
+        fields:
+            description: Comma seperated string providing additional table columns to add as host vars to each inventory host.
+            type: list
+            default: []
+        sn_groups:
+            description: Comma seperated string providing additional table columns to use as groups. Groups can overlap with fields
+            type: list
+            default: []
+        selection_order:
+            description: Comma seperated string providing ability to define selection preference order.
+            type: list
+            default: 'host_name,fqdn,ip_address'
+        filter_results:
+            description: Filter results with sysparm_query encoded query string syntax. Complete list of operators available for filters and queries.
+            type: string
+            default: ''
+        proxy:
+            description: Proxy server to use for requests to ServiceNow.
+            type: string
+            default: ''
+'''
+
+EXAMPLES = r'''
+plugin: now
+instance: demo.service-now.com
+username: admin
+password: password
+'''
+
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, to_safe_group_name, Cacheable
+from ansible.errors import AnsibleError, AnsibleParserError
+try:
+    import requests
+except ImportError:
+    raise AnsibleError('This script requires python-requests')
```

Comment:
Please use this kind of pattern for requirements https://github.com/ansible/ansible/blob/bbbaf0dfe775bb7ab8584244a7162cd3a9a5f9a5/lib/ansible/plugins/inventory/vmware_vm_inventory.py#L109-L114 and then handle missing libraries in parse() like this https://github.com/ansible/ansible/blob/bbbaf0dfe775bb7ab8584244a7162cd3a9a5f9a5/lib/ansible/plugins/inventory/vmware_vm_inventory.py#L216-L218